soc/nxp/imxrt/imxrt118x/Kconfig

```diff
@@ -0,0 +1,63 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_IMXRT118X
+	select CPU_CORTEX_M_HAS_DWT
+	select PLATFORM_SPECIFIC_INIT
+	select ARM
+	select CLOCK_CONTROL
+	select HAS_MCUX_CACHE
+	select ARMV8_M_DSP if SOC_MIMXRT1189_CM33
+	select CPU_HAS_ARM_SAU if SOC_MIMXRT1189_CM33
+	select HAS_MCUX
+	select HAS_MCUX_CCM_REV2
+	select HAS_MCUX_LPUART
+	select CPU_HAS_ARM_MPU
+	select INIT_ARM_PLL
+	select ARM_TRUSTZONE_M if SOC_MIMXRT1189_CM33
+	select CPU_HAS_ICACHE if SOC_MIMXRT1189_CM7
+	select CPU_HAS_DCACHE if SOC_MIMXRT1189_CM7
+	select CPU_HAS_FPU
+	select CPU_HAS_FPU_DOUBLE_PRECISION if SOC_MIMXRT1189_CM7
+	select HAS_MCUX_IOMUXC
+	select HAS_SWO
+
+config SOC_MIMXRT1189_CM33
+	select CPU_CORTEX_M33
+
+config SOC_MIMXRT1189_CM7
+	select CPU_CORTEX_M7
+
+if SOC_SERIES_IMXRT118X
+
+config MCUX_CORE_SUFFIX
+	default "_cm7" if SOC_MIMXRT1189_CM7
+	default "_cm33" if SOC_MIMXRT1189_CM33
+
+config BYPASS_LDO_LPSR
+	bool "Bypass LDO lpsr"
+
+config ADJUST_LDO
```

Comment:
Because the architecture of the entire RT118X is different from that of RT11XX, it is not compatible. For example, RT11XX is a dual-core chip combined with CM7 and CM4, but RT118X is a dual-core chip combined with CM33 and CM7, and CM33 can only be the master core

Reply:
> these configs are copied from the RT11XX. We should reuse code instead of copying. Also, why can't RT118x be in the RT11xx folder for more code and config reuse.

These kconfig don't be used on RT1180 soc.c file, I will delete them.